src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupLinearFunctional.java

```diff
@@ -0,0 +1,516 @@
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public AColGroup copy() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public boolean containsValue(double pattern) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public long getNumberNonZeros(int nRows) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public AColGroup replace(double pattern, double replace) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public void readFields(DataInput in) throws IOException {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public void write(DataOutput out) throws IOException {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public long getExactSizeOnDisk() {
+		long ret = super.getExactSizeOnDisk();
+		ret += _coefficents[0].length * _coefficents.length * 8L;
+		ret += 4L; // _numRows
+		return ret;
+	}
+
+	@Override
+	protected void computeProduct(double[] c, int nRows) {
+		c[0] = 1;
+		for(int colIndex : _colIndexes) {
+			double intercept = _coefficents[0][colIndex];
+			double slope = _coefficents[1][colIndex];
+			for(int i = 0; i < nRows; i++) {
+				c[0] *= intercept + slope * (i + 1);
+			}
+		}
+	}
+
+	@Override
+	protected void computeRowProduct(double[] c, int rl, int ru, double[] preAgg) {
+		for(int rix = rl; rix < ru; rix++) {
+			c[rix] = 1;
+			for(int colIndex : _colIndexes) {
+				double intercept = _coefficents[0][colIndex];
+				double slope = _coefficents[1][colIndex];
+				c[rix] *= intercept + slope * (rix + 1);
+			}
+		}
+	}
+
+	@Override
+	protected void computeColProduct(double[] c, int nRows) {
+		for(int colIndex : _colIndexes) {
+			c[colIndex] = 1;
+			double intercept = _coefficents[0][colIndex];
+			double slope = _coefficents[1][colIndex];
+			for(int i = 0; i < nRows; i++) {
+				c[colIndex] *= intercept + slope * (i + 1);
+			}
+		}
+	}
+
+	@Override
+	protected double[] preAggSumRows() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	protected double[] preAggSumSqRows() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	protected double[] preAggProductRows() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	protected double[] preAggBuiltinRows(Builtin builtin) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public long estimateInMemorySize() {
+		return ColGroupSizes.estimateInMemorySizeLinearFunctional(getNumCols());
+	}
+
+	@Override
+	public CM_COV_Object centralMoment(CMOperator op, int nRows) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public AColGroup rexpandCols(int max, boolean ignore, boolean cast, int nRows) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double getCost(ComputationCostEstimator e, int nRows) {
+		final int nCols = getNumCols();
+		// We store 2 tuples in this column group, namely intercepts and slopes
+		return e.getCost(nRows, nRows, nCols, 2, 1.0);
```

Comment:
this is going to be tricky to calculate, since this does not correctly reflect the calculation.
i suggest you add a 
`LOG.warn("Not precise cost calculation for LinearFunctional ColGroup");`